src/sentry/dynamic_sampling/tasks.py

```diff
@@ -145,55 +172,53 @@ def adjust_sample_rates(
     """
-    with metrics.timer("sentry.tasks.dynamic_sampling.process_projects_factors.core"):
-        redis_client = get_redis_client_for_ds()
-        Counter = namedtuple("Counter", ["count", "count_keep", "count_drop"])
-        project_ids_with_counts = {}
-        for project_id, count_per_root, count_keep, count_drop in projects_with_tx_count_and_rates:
-            project_ids_with_counts[project_id] = Counter(count_per_root, count_keep, count_drop)
-
-        with redis_client.pipeline(transaction=False) as pipeline:
-            for project in Project.objects.get_many_from_cache(project_ids_with_counts.keys()):
-                desired_sample_rate = quotas.get_blended_sample_rate(project)
-
-                if not desired_sample_rate:
-                    continue
-
-                counter = project_ids_with_counts[project.id]
-                if (actual_rate := actual_sample_rate(counter.count_keep, counter.count_drop)) != 0:
-                    adj_factor_cache_key = generate_cache_key_adj_factor(org_id)
-
-                    try:
-                        prev_factor = float(pipeline.hget(adj_factor_cache_key, project.id))
-                    except (TypeError, ValueError):
-                        prev_factor = 1.0
-
-                    new_factor = adjusted_factor(prev_factor, actual_rate, desired_sample_rate)
-
-                    pipeline.hset(adj_factor_cache_key, project.id, new_factor)
-                    pipeline.pexpire(adj_factor_cache_key, CACHE_KEY_TTL)
-
-                schedule_invalidate_project_config(
-                    project_id=project.id, trigger="dynamic_sampling_process_projects_factors"
-                )
-            pipeline.execute()
+
+    redis_client = get_redis_client_for_ds()
+    factor_key = generate_cache_key_rebalance_factor(org_volume.org_id)
+    # we need a project from the current org... unfortunately get_blended_sample_rate
+    # takes a project (not an org)
+    # TODO RaduW is there a better way to get an org Project than filtering ?
+    org_projects = Project.objects.filter(org_id=org_volume.org_id)
+    for project in org_projects:
+        desired_sample_rate = quotas.get_blended_sample_rate(project)
+        break
+    else:
+        redis_client.delete(factor_key)  # cleanup just to be sure
+        # org with no project this shouldn't happen
+        return "no project found"
+
+    if org_volume.total == 0 or org_volume.indexed == 0:
+        # not enough info to make adjustments ( we don't consider this an error)
+        return None
+
+    previous_interval_sample_rate = org_volume.indexed / org_volume.total
+    try:
+        previous_factor = float(redis_client.get(factor_key))
+    except (TypeError, ValueError):
+        previous_factor = 1.0
+
+    new_factor = adjusted_factor(
+        previous_factor, previous_interval_sample_rate, desired_sample_rate
+    )
+
+    if new_factor < 0.1 or new_factor > 10:
```

Comment:
Can we put these two numbers as `CLAMP_MIN` and `CLAMP_MAX`? I would like to highlight them with a proper comment.